mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -1086,99 +1018,77 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
                     </div>
                 }
             </div>
-            {agentModeEnabled && displayOptimizedChatHistory.some(msg =>
-                msg.type === 'openai message:agent:planning'
-            ) ? (
-                <div className="agent-controls">
-                    <button
-                        className="button-base button-purple agent-start-button"
-                        onClick={executeAgentPlan}
-                    >
-                        Let&apos;s go!
-                    </button>
-                    <button
-                        className="button-base button-red agent-cancel-button"
-                        onClick={async () => {
+            <ChatInput
+                initialContent={''}
+                placeholder={
+                    agentExecutionStatus === 'working' ? 'Agent is working...' :
+                        agentExecutionStatus === 'stopping' ? 'Agent is stopping...' :
+                            agentModeEnabled ? 'Ask agent to do anything' :
+                                displayOptimizedChatHistory.length < 2 ? `Ask question (${operatingSystem === 'mac' ? '⌘' : 'Ctrl'}E), @ to mention`
+                                    : `Ask followup (${operatingSystem === 'mac' ? '⌘' : 'Ctrl'}E), @ to mention`
+                }
+                onSave={agentModeEnabled ? startAgentExecution : sendChatInputMessage}
+                onCancel={undefined}
+                isEditing={false}
+                contextManager={contextManager}
+                notebookTracker={notebookTracker}
+                renderMimeRegistry={renderMimeRegistry}
+                agentModeEnabled={agentModeEnabled}
+            />
+            {agentExecutionStatus !== 'working' && agentExecutionStatus !== 'stopping' && (
+                <div className="chat-controls">
+                    <ToggleButton
+                        leftText="Chat"
+                        rightText="Agent"
+                        isLeftSelected={!agentModeEnabled}
+                        onChange={async (isLeftSelected) => {
                             await startNewChat(); // TODO: delete thread instead of starting new chat
-                            setAgentModeEnabled(false);
+                            console.log('Calling this On Change function')
```

Comment:
Delete

Reply:
Keep eye out for a bug where agent mode is getting turned off when it should not be?